core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -45,173 +45,203 @@ oppia.directive('schemaBasedListEditor', [
-            $scope.showAddItemButton();
-          };
+            if ($scope.localValue.length === 1) {
+              if ($scope.localValue[0].length === 0) {
+                $scope.isAddItemButtonPresent = false;
+              }
+            }
 
-          $scope.showAddItemButton = function() {
-            deleteEmptyElements();
-            $scope.isAddItemButtonPresent = true;
-          };
+            $scope.lastElementOnBlur = function() {
+              _deleteLastElementIfUndefined();
+              $scope.showAddItemButton();
+            };
 
-          $scope.hideAddItemButton = function() {
-            $scope.isAddItemButtonPresent = false;
-          };
+            $scope.showAddItemButton = function() {
+              deleteEmptyElements();
+              $scope.isAddItemButtonPresent = true;
+            };
 
-          $scope._onChildFormSubmit = function(evt) {
-            if (!$scope.isAddItemButtonPresent) {
-              /**
-               * If form submission happens on last element of the set (i.e the
-               * add item button is absent) then automatically add the element
-               * to the list.
-               */
-              if (($scope.maxListLength === null ||
-                   $scope.localValue.length < $scope.maxListLength) &&
-                  !!$scope.localValue[$scope.localValue.length - 1]) {
-                $scope.addElement();
+            $scope.hideAddItemButton = function() {
+              $scope.isAddItemButtonPresent = false;
+            };
+
+            $scope._onChildFormSubmit = function(evt) {
+              if (!$scope.isAddItemButtonPresent) {
+                /**
+                 * If form submission happens on last element of the set (i.e
+                 * the add item button is absent) then automatically add the
+                 * element to the list.
+                 */
+                if (($scope.maxListLength === null ||
+                    $scope.localValue.length < $scope.maxListLength) &&
+                    !!$scope.localValue[$scope.localValue.length - 1]) {
+                  $scope.addElement();
+                }
+              } else {
+                /**
+                 * If form submission happens on existing element remove focus
+                 * from it
+                 */
+                document.activeElement.blur();
+              }
+              evt.stopPropagation();
+            };
+
+            var LARGE_NUMBER = 99;
+            var answerGroups = ResponsesService.getAnswerGroups();
+            var answerChoices = ResponsesService.getAnswerChoices();
+
+            var reduceRuleIndexByOne = function(answerGroupIndex, ruleIndex) {
+              answerGroups[answerGroupIndex].rules[ruleIndex].inputs.x--;
+              return answerGroups;
+            };
+
+            var makeRuleInvalid = function(answerGroupIndex, ruleIndex) {
+              answerGroups[answerGroupIndex].rules[ruleIndex].inputs.x =
```

Comment:
why not set this to something like -1?

Reply:
Yes, I did think of that but rules are defined as non-negative integers in the codebase and making it ```undefined``` causes the site to break!

Reply:
okay, this should be fine then.